index.js

```diff
@@ -31,6 +31,13 @@ function wasmify (b, options) {
           buf[b++] = ((third & 3) << 6) | (table[src.charCodeAt(i+3)] & 63)
         }
 
+        if (imports) {
+          // return the the asynchronous WebAssembly.instantiate(buf) since this
+          // is the primary API for compiling and instantiating WebAssembly code
+          // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate
+          return WebAssembly.instantiate(buf)
```

Comment:
Doesn't look like you are passing the `imports` here.

I think we should also be consistent with the options and provide sync instance:

```js
return sync ? new WebAssembly.Instance(buf, imports) : WebAssembly.instance(buf, imports)
```

For WASM tiny modules, this would allow a direct import:

```js
var sample = require('./sample.wasm')

sample({ ...imports })
```

Reply:
I ended up using this in `rollup-plugin-wasm`:

```js
        if (imports && !sync) {
          return WebAssembly.instantiate(buf, imports)
        } else if (!imports && !sync) {
          return WebAssembly.compile(buf)
        } else {
          var mod = new WebAssembly.Module(buf)
          return imports ? new WebAssembly.Instance(mod, imports) : mod
        }
```

Reply:
OK @jamen it seems I forgot to push the imports one. Will push another commits for this. Thanks